src/API/Google/Middleware.php

```diff
@@ -613,4 +618,34 @@ public function account_request_review( $regions ) {
 			);
 		}
 	}
+
+	/**
+	 * This function detects if the current account is a sub-account
+	 * This function is cached in the MC_IS_SUBACCOUNT transient
+	 *
+	 * @return bool True if it's a standalone account.
+	 */
+	public function is_subaccount(): bool {
+		/** @var TransientsInterface $transients */
+		$transients    = $this->container->get( TransientsInterface::class );
+		$is_subaccount = $transients->get( $transients::MC_IS_SUBACCOUNT );
+
+		if ( is_null( $is_subaccount ) ) {
+			$is_subaccount = 0;
+
+			$merchant_id = $this->options->get_merchant_id();
+			$accounts    = $this->get_merchant_accounts();
+
+			foreach ( $accounts as $account ) {
+				if ( $account['id'] === $merchant_id && $account['subaccount'] ) {
+					$is_subaccount = 1;
+				}
+			}
+
+			$transients->set( $transients::MC_IS_SUBACCOUNT, $is_subaccount );
+		}
+
+		// since transients doesn't support booleans, we save them as 0/1 and do the conversion here
```

Comment:
```suggestion
		// since transients don't support booleans, we save them as 0/1 and do the conversion here
```

I've seen other solutions and in WC where the values "yes" / "no" are saved to get around this problem, but integer values work as well.